File: netlogo-gui/src/main/prim/_goto.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.core.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;

public final strictfp class _goto  // , Matt
    extends Command {
  public _goto(int offset) {
    this.offset = offset;
  }



  @Override
  public String toString() {
    return super.toString() + ":" + offset;
  }

  @Override
  public void perform(final Context context) {
    context.ip = offset;
  }

  public void perform_1(final Context context) {
    context.ip = offset;
  }
}

File: netlogo-gui/src/main/prim/_goto.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.nvm.{ Command, Context }

class _goto extends Command {

  override def toString: String = super.toString + ":" + offset

  override def perform(context: Context): Unit = perform_1(context)

  def perform_1(context: Context): Unit = context.ip = offset
}
